daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/ExpressionEvaluatingParsers.scala

```diff
@@ -165,7 +165,11 @@ class NewVariableInstanceStartParser(override val context: VariableRuntimeData)
   override lazy val runtimeDependencies = Vector()
 
   def parse(start: PState): Unit = {
-    start.newVariableInstance(context)
+    if (context.maybeDefaultValueExpr.isDefined) {
+      val res = context.maybeDefaultValueExpr.get.evaluate(start)
+      start.newVariableInstance(context, DataValue.unsafeFromAnyRef(res))
+    } else
+      start.newVariableInstance(context)
```

Comment:
Benefit of above suggestion about splitting up NVI and default value, this becomes something like
```scala
start.newVariableInstance(context)
if (context.maybeDefaultVlueExpr.isDefined) {
  val res = context.maybeDefaultValueExpr.get.evaluate(start)
￼  start.setVariableDefault(...)
}
```
Benefit here is it decouples creating a new variable instance and setting the default. Granted the two are almost always going to be coupled, but it makes it easier to set what state setting a default value changes.